test/unit/test_utils.py

```diff
@@ -351,3 +353,32 @@ def test_split(num_factors):
 
 def test_get_num_gpus():
     assert utils.get_num_gpus() >= 0
+
+
+def _touch_file(fname, compressed: bool, empty: bool) -> str:
+    if compressed:
+        open_func = gzip.open
+    else:
+        open_func = open
+    with open_func(fname, encoding='utf8', mode='wt') as f:
+        if not empty:
+            for i in range(10):
+                print(str(i), file=f)
+    return fname
+
+
+def test_is_gzip_file():
+    with TemporaryDirectory() as temp:
+        fname = os.path.join(temp, 'test')
+        assert utils.is_gzip_file(_touch_file(fname, compressed=True, empty=True))
+        assert utils.is_gzip_file(_touch_file(fname, compressed=True, empty=False))
+        assert not utils.is_gzip_file(_touch_file(fname, compressed=False, empty=True))
+        assert not utils.is_gzip_file(_touch_file(fname, compressed=False, empty=False))
+
+
+def test_smart_open_without_suffix():
+    with TemporaryDirectory() as temp:
+        fname = os.path.join(temp, 'test')
+        _touch_file(fname, compressed=True, empty=False)
```

Comment:
would it make sense to add the same test with `compressed=False`?

Reply:
sure, will add